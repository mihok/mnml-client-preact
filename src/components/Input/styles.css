/* Input Component */


/* Messenger Theme */

.Input__form--messenger {
  display: flex !important;
  min-height: 40px;
  background: white;
  border-right: 1px solid rgba(0,0,0,0.1);
}

.Input--messenger {
  padding: 12px;
  outline: 0;
  border: 0px;
  flex-grow: 1;
}


/* Float Theme */

.Input--float {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 48px;
  box-sizing: border-box;
  box-shadow: #dddddd 1px 1px 8px 0;
  border: 0;
  background-color: #ffffff;
  border-radius: 10px;
  padding: 16px;
  color: #222222;
  font-size: 13px;
  resize: none;
  outline: 0;
}

.Input--float:disabled {
  background-color: #e1e1e1;
}
