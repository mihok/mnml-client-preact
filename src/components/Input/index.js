import { h, render, Component } from "preact";
import "./styles.css";
import theme from '../ThemeProvider/themeHOC';

const Input = props => {
  const {sendMessage, theme, textBox, handleInput} = props
  return (
    <form class={`Input__Form__${theme}`} onSubmit={sendMessage}>
      <input
        class={`Input__${theme}`}
        placeholder="Type Here"
        onChange={e => handleInput(e)}
        name="messages"
        value={textBox}
      />
    </form>
  );
};
export default theme(Input);
