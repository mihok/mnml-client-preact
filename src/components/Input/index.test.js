/* const render = require('preact-render-to-string');
 * const App = require('../thing');*/
import render from 'preact-render-to-string';
import { h } from 'preact';
import Input from '.';

describe('App component', () => {
  it('should render ', () => {
    const tree = render(<Input />);
    expect(tree).toMatchSnapshot();
  });
});
