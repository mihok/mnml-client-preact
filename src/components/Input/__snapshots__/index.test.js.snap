// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render  1`] = `"<form class=\\"Input__Form__undefined\\"><input class=\\"Input__undefined\\" placeholder=\\"Type Here\\" name=\\"messages\\" /></form>"`;
